### PR TITLE
[Reading Challenge Widget] Fixes missing behavior logic

### PR DESCRIPTION
**Pull request by Williamrai**
### What does this do?
- adds ongoing streak logic post end date
- Add `hasActiveStreak` field to `ReadingChallengeUserData` to distinguish active vs. broken streaks
- Prevent streak resets and article-read updates once the goal is reached
- Fix `isChallengeActive` to be inclusive of END_DATE
- Non-enrolled users past END_DATE now resolve to `ChallengeConcludedNoStreak` instead of `NotEnrolled`
- Replace tinted flame icon with new `ic_streak_warning` drawable for the "needs reading" badge state
- Make `WidgetBadge` `iconTintColor` optional (nullable)
- Update and expand unit tests to cover buffer period edge cases

**Phabricator:**
https://phabricator.wikimedia.org/T418641
https://phabricator.wikimedia.org/T418626

### CODE REVIEW

File: app/src/main/java/org/wikipedia/widgets/readingchallenge/ReadingChallengeWidget.kt [hunk 1]
```diff
@@ -138,7 +138,7 @@ fun WidgetBadge(
             contentDescription = null,
             modifier = GlanceModifier
                 .size(iconSize),
-            colorFilter = ColorFilter.tint(ColorProvider(day = iconTintColor, night = iconTintColor))
+            colorFilter = if (iconTintColor != null) ColorFilter.tint(ColorProvider(day = iconTintColor, night = iconTintColor)) else null
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/widgets/readingchallenge/ReadingChallengeWidget.kt]: Maybe use `iconTintColor?.let { ColorFilter.tint(ColorProvider(day = it, night = it)) }` is simpler

Comment (Williamrai) [reply app/src/main/java/org/wikipedia/widgets/readingchallenge/ReadingChallengeWidget.kt]: done.

**Comment by Williamrai** [comment]
Thank you @cooltey. The new layout changes looks good.
